### PR TITLE
TechDocs: Inject CSS transformer and initial backstage style integration for reader

### LINKED ISSUE #2399: TechDocs: Links just look like text
## Expected Behavior

Links are obvious

## Current Behavior

Links just look like text
image

cc @emmaindal 

## Possible Solution

Style it

## Steps to Reproduce

Add a link inside your static site.

1. Step 1
2. Step 2
3. ...

## Your Environment

- NodeJS Version (v12):
- Operating System and Version (e.g. Ubuntu 14.04):
- Browser Information:

### LINKED ISSUE #2463: TechDocs to use the Backstage theme for fonts

### LINKED ISSUE #2564: TechDocs: Create a new inject css transformer

### PR DESCRIPTION
**Pull request by hooloovooo**
## Hey, I just made a Pull Request!

This adds a injectCss transformer and contains a suggestion for approach to integrating with the backstage theme.

Looking for feedpack on injectCss and theme integration. Will make the integration tighter when we've settled on how it should work.

Closes #2564
Closes #2399

2020-09-23-111439_1072x611_scrot

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

File: plugins/techdocs/src/reader/transformers/injectCss.test.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createTestShadowDom } from '../../test-utils';
+import { injectCss } from '../transformers';
+
+describe('rewriteDocLinks', () => {
```
Comment (emmaindal) [inline plugins/techdocs/src/reader/transformers/injectCss.test.ts:20]:
```suggestion
describe('injectCss', () => {
```

File: plugins/techdocs/src/reader/transformers/injectCss.test.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  it('should not do anything', () => {
```
Comment (emmaindal) [inline plugins/techdocs/src/reader/transformers/injectCss.test.ts:21]: `should not do anything` can we clarify this?

**Comment by emmaindal** [comment]
I think this is a good start. I can see the injectCSS transformer grow and that we need to separate it out to maybe inject a whole stylesheet link instead but for now, imo, I think we can go with this.

**Comment by emmaindal** [comment]
closes #2463

**Review by emmaindal — APPROVED** [review]

**Comment by hooloovooo** [comment]
Currently working on mapping some on the non working css vars in :root to the backstage theme. Merging this now and we can keep improving the integration as we find things that doesn't look correct.

**Pull request merged**